### PR TITLE
SELinux blog post

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
Blog article about SELinux in KubeVirt

**Does this PR fix any issue?** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:

> Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: _posts/2020-05-25-SELinux-from-basics-to-KubeVirt.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+Another application that leverages MCS is Linux containers.  
+In fact, containers use very few SELinux types and rely mostly on MCS to provide container isolation.  
+For example, all the files and processes in container filesystems have the same SELinux types.  
+For a non-super-privileged container, those types are usually container_file_t for file and container_t for processes.  
+Most operations are permitted within those types, and the categories are really what matters.  
+As with libvirt, categories have to match for access to be granted, effectively blocking inter-container communication.
+
+Super-privileged containers however are exempt from categories. They use the spc_t SELinux type, which allows them to do pretty much anything, at least as far as SELinux is concerned.  
+That is all defined as an SELinux module in https://github.com/containers/container-selinux
```
Comment (codificat) [inline _posts/2020-05-25-SELinux-from-basics-to-KubeVirt.md:82]: Same as above with the URL: either surround with angle brackets to autolink, or create a link within the text.

File: _posts/2020-05-25-SELinux-from-basics-to-KubeVirt.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Note: MCS can do a lot more, this article only describes the bits that are used by libvirt and kubernetes.
+
+### MCS and containers
+
+Another application that leverages MCS is Linux containers.  
+In fact, containers use very few SELinux types and rely mostly on MCS to provide container isolation.  
+For example, all the files and processes in container filesystems have the same SELinux types.  
+For a non-super-privileged container, those types are usually container_file_t for file and container_t for processes.  
+Most operations are permitted within those types, and the categories are really what matters.  
+As with libvirt, categories have to match for access to be granted, effectively blocking inter-container communication.
```
Comment (codificat) [inline _posts/2020-05-25-SELinux-from-basics-to-KubeVirt.md:79]: Blank lines to separate paragraphs and no line breaks within the same paragraph.

backtick quotes for types

File: _posts/2020-05-25-SELinux-from-basics-to-KubeVirt.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+Super-privileged containers however are exempt from categories. They use the spc_t SELinux type, which allows them to do pretty much anything, at least as far as SELinux is concerned.  
+That is all defined as an SELinux module in https://github.com/containers/container-selinux
+
+### MCS and container orchestrators
+
+Container orchestrators add a level of management.  
+They define pods of containers, and within a pod, cross-container communication is acceptable and often even necessary.  
+Categories are therefore managed at the pod level, and all the containers that belong to the same pod are assigned the same categories, as illustrated by the following diagram.
```
Comment (codificat) [inline _posts/2020-05-25-SELinux-from-basics-to-KubeVirt.md:88]: Single paragraph? Then no line breaks

File: _posts/2020-05-25-SELinux-from-basics-to-KubeVirt.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+![Components View](/assets/2020-05-25-SELinux-from-basics-to-KubeVirt/kubernetes.svg)
+
+## SELinux in Kubevirt
+
+Finally getting to KubeVirt, which relies on all of the above, as it runs libvirt in a container managed by a container orchestrator on SELinux-enabled systems.
```
Comment (codificat) [inline _posts/2020-05-25-SELinux-from-basics-to-KubeVirt.md:94]: blank line

**Comment by codificat** [comment]
Thanks for the blog post submission @jean-edouard.  I commented about a few formatting suggestions.

Besides that: the spell checker is complaining about [a few unknown words](https://travis-ci.org/github/kubevirt/kubevirt.github.io/builds/689642041#L380-L395). Please add them to `.yaspeller.json`.

**Comment by jean-edouard** [comment]
Thanks a lot @codificat for the detailed review!
Force-pushed to address the comments.

**Comment by codificat** [comment]
Thanks for the update.

The spell checker is [still complaining](https://travis-ci.org/github/kubevirt/kubevirt.github.io/builds/690043693#L383-L387) - you'd have to add your name and surname to begin with, but there's more: it is mixing up words and thinking that "The SELinux" actually meant "These Linux" and similar with MCS... I believe we'll have to fine tune that spellchecker :/

**Comment by fabiand** [comment]
@codificat what needs to be done to teach the spellchecker?

**Comment by ptrnull** [comment]
> @codificat what needs to be done to teach the spellchecker?

@fabiand @codificat  line 135 in https://github.com/kubevirt/community/pull/43/files
That PR should be merged to have that info published and available to everybody.

**Comment by codificat** [comment]
> > @codificat what needs to be done to teach the spellchecker?
> 
> @fabiand @codificat line 135 in https://github.com/kubevirt/community/pull/43/files
> That PR should be merged to have that info published and available to everybody.

Unfortunately that's not enough for the conext-aware complaints here.

The only way I found so far is to add the `--by-words` option to the spell checker invocation.

I'm doing some more tests and will send a PR to fix the spellchecker configuration (and update it to the current version). I'll update kubevirt/community#43 too.

**Comment by codificat** [comment]
I believe that jean-edouard/kubevirt.github.io#1 should fix the issues with the spell checker and allow this PR to move forward.

If you prefer I can submit that as a separate PR and then ask you to rebase yours after merge.

**Comment by jean-edouard** [comment]
Thank you @codificat! I merged your PR, and rebased the branch. I was first trying to rebase against master, hence the mess... Should be ok now.

**Comment by codificat** [comment]
booh - the CI test is still failing due to the [spell checker|https://travis-ci.org/github/kubevirt/kubevirt.github.io/builds/691736692#L381-L385] :( 

@jean-edouard could it be that the rebase skipped the changes in yaspeller configuration?

Sorry about this mess - if you prefer I can go the other route of submitting the spell checker changes through a separate PR and ask you to rebase afterwards

**Comment by jean-edouard** [comment]
@codificat it's my bad, I re-added your commit...

**Comment by codificat** [comment]
Yay! looking good now.

/lgtm
/approve

thanks for the contribution!

**Pull request merged**